Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDllHelpers/DebugCallbacks.cs

```diff
@@ -0,0 +1,109 @@
+﻿using DbgEngManaged;
+using System;
+
+namespace CsDebugScript.Engine.Debuggers.DbgEngDllHelpers
+{
+    /// <summary>
+    /// Debug callbacks called during WaitForEvent callback.
+    /// This class in future can be extended to support callbacks provided
+    /// on certain actions (e.g. breakpoint hit, thread create, module load etc.)
+    /// </summary>
+    class DebugCallbacks : IDebugEventCallbacks
+    {
+        /// <summary>
+        /// IDebugClient.
+        /// </summary>
+        private IDebugClient client;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="client">IDebugClient interface.</param>
+        /// <param name="debugStatusGoEvent">Event used to signal when debuggee switches to release state.</param>
+        public DebugCallbacks(IDebugClient client, System.Threading.AutoResetEvent debugStatusGoEvent)
+        {
+            this.client = client;
+            this.debugStatusGoEvent = debugStatusGoEvent;
+            this.client.SetEventCallbacks(this);
+        }
+
+        /// <summary>
+        /// Event used to signal go state. Used to unblock DebugFlow loop.
+        /// </summary>
+        private System.Threading.AutoResetEvent debugStatusGoEvent;
+
+        public void Breakpoint(IDebugBreakpoint Bp)
+        {
+            throw new NotImplementedException();
```

Comment:
Yes, I am putting NotImplemented to all the callbacks (-ChangeDebugeeState). In next push request I will provide breakpoint support.

---
In reply to: [69983189](https://github.com/southpolenator/WinDbgCs/pull/8#discussion_r69983189) [](ancestors = 69983189)